source/dub/project.d

```diff
@@ -1161,12 +1161,41 @@ private string processVars(string var, in Project project, in Package pack, bool
 		vres.put(var);
 		var = vres.data;
 	}
-	if (is_path) {
-		auto p = Path(var);
-		if (!p.absolute) {
-			return (pack.path ~ p).toNativeString();
-		} else return p.toNativeString();
-	} else return var;
+
+	static if (glob)
+		assert(is_path, "can't glob something that isn't a path");
+	else {
+		if (!is_path)
+			return var;
+	}
+	auto p = Path(var);
+	string res;
+	if (!p.absolute)
+		res = (pack.path ~ p).toNativeString();
+	else
+		res = p.toNativeString();
+	static if (!glob)
+		return res;
+	else {
+		// Find the unglobbed prefix and iterate from there.
+		size_t i = 0;
+		size_t sepIdx = 0;
+		loop: while (i < res.length) {
+			switch_: switch (res[i])
+			{
+			case '*', '?', '[', '{': break loop;
+			case '/': sepIdx = i; goto default;
+			default: ++i; break switch_;
+			}
+		}
+		if (i == res.length) //no globbing found in the path
+			return [res];
+		import std.path : globMatch;
+		return dirEntries(res[0 .. sepIdx], SpanMode.depth)
+			.map!(de => de.name)
+			.filter!(name => globMatch(name, res))
```

Comment:
This will iterate over all files and not stop if the folder is not included in the glob - in other words it's not globing, more filtering with a regex ;-)